### PR TITLE
Add redux snippets

### PR DESCRIPTION
**Pull request by chris542**
I created some snippets for redux which I find really annoying to type.
If anyone wants to change the way I wrote the snippet, feel free to do so

### CODE REVIEW

File: snippets/javascript/javascript-redux.snippets [hunk 1]
```diff
@@ -0,0 +1,37 @@
+snippet ist
+	import { createStore } from 'redux'
+snippet con
+	connect(${1:mapStateToProps},${2:mapDispatchToProps})(<${3:VISUAL}/>)
```
Comment (lpil) [inline snippets/javascript/javascript-redux.snippets]: Could you add a space after the comma please?

File: snippets/javascript/javascript-redux.snippets [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+snippet act
+	const ${1:actionName} = (${2:arg}) => {
+		return {
+			type: ${3:VISUAL},
+			$2
+		}
+	}
+snippet rdc
+	const ${1:reducerName} = (state={},action) =>{
```
Comment (lpil) [inline snippets/javascript/javascript-redux.snippets]: Here also for a space after the comma please

File: snippets/javascript/javascript-redux.snippets [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+		switch(action.type){
```
Comment (lpil) [inline snippets/javascript/javascript-redux.snippets]: Space before `{` please

File: snippets/javascript/javascript-redux.snippets [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+			case ${1:action}:
+				return {
+					...state,
+					$2
+				}
+			default :
```
Comment (lpil) [inline snippets/javascript/javascript-redux.snippets]: No space after `default` please

File: snippets/javascript/javascript-redux.snippets [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+				return state
+		}
+	}
+snippet mstp
+	const mapStateToProps = (state)=>{
```
Comment (lpil) [inline snippets/javascript/javascript-redux.snippets]: Spaces around `=>` please

File: snippets/javascript/javascript-redux.snippets [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		return {
+			${1:propName} : state.$1,
```
Comment (lpil) [inline snippets/javascript/javascript-redux.snippets]: No space before `:` please

File: snippets/javascript/javascript-redux.snippets [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+		}
+	}
+snippet mdtp
+	const mapDispatchToProps = (dispatch)=>{
```
Comment (lpil) [inline snippets/javascript/javascript-redux.snippets]: Spaces around `=>` please

File: snippets/javascript/javascript-redux.snippets [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+		return {
+			${1:propName} : ()=>{
```
Comment (lpil) [inline snippets/javascript/javascript-redux.snippets]: No space before `:` and spaces around `=>` please

**Comment by chris542** [comment]
Hey @lpil,

Made fixes as requested :)

File: snippets/javascript/javascript-redux.snippets [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+	const mapStateToProps = (state) => {
+		return {
+			${1:propName}: state.$1,
+		}
+	};
+snippet mdtp
+	const mapDispatchToProps = (dispatch) => {
+		return {
+			${1:propName}: () => {
+				dispatch(${2:actionName}())
```
Comment (lpil) [inline snippets/javascript/javascript-redux.snippets]: Could we get a `;` after this function call please? Thanks

**Comment by chris542** [comment]
Hey @lpil, made those changes :)
Hopefully i can get rid of my fork now!! haha

File: snippets/javascript/javascript-redux.snippets [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+snippet rdc
+	const ${1:reducerName} = (state={}, action) => {
+		switch(action.type) {
+			case ${1:action}:
+				return {
+					...state,
+					$2
+				};
+			default:
+				return state
```
Comment (lpil) [inline snippets/javascript/javascript-redux.snippets]: This one too :)

**Comment by chris542** [comment]
Maybe I shouldn't have force-pushed changes. Hmm. But I added ; after return state on line 21 :)

**Pull request merged**